File: js/leetcode/happy-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function (n) {
  const initialNumberList = [];
  number = n;
  while (number != 1) {
    n = number;
    number = 0;
    while (n != 0) {
      number += Math.pow(n % 10, 2);
      n = Math.floor(n / 10);
    }

    if (initialNumberList.includes(number)) {
      break;
    }

    initialNumberList.push(number);
  }
  return number == 1 ? true : false;
};

module.exports = isHappy;

File: leetcode-75-level-2/happy-number.py

class Solution:
    def isHappy(self, n: int) -> bool:
        cycle = []
        number = n
        while number != 1:
            number = 0
            while n:
                number += (n%10) ** 2
                n = n // 10

            if number in cycle:
                break
            cycle.append(number)
            n = number
        
        return True if number==1 else False

